framework/src/mesh/ConcentricCircleMesh.C

```diff
@@ -0,0 +1,660 @@
+
+  counter = 0;
+
+  while (index < limit)
+  {
+
+    Elem * elem = mesh.add_elem(new Quad4);
+    elem->set_node(0) = nodes[index];
+    elem->set_node(1) = nodes[index + _num_sectors + 1];
+    elem->set_node(2) = nodes[index + _num_sectors + 2];
+    elem->set_node(3) = nodes[index + 1];
+
+    for (int i = 0; i < static_cast<int>(subdomainIDs.size()) - 1; ++i)
+    {
+      if (index < limit - (standard + 1) * i && index >= limit - (standard + 1) * (i + 1))
+        elem->subdomain_id() = subdomainIDs.at(subdomainIDs.size() - 1 - i);
+    }
+
+    int const initial = std::pow(standard / 2 + 1, 2);
+    int const final = std::pow(standard / 2 + 1, 2) + _num_sectors - 1;
+
+    if ((index - initial) % (standard + 1) == 0)
+      boundary_info.add_side(elem, 0, 2);
+    if ((index - final) % (standard + 1) == 0)
+      boundary_info.add_side(elem, 2, 1);
+    if (index > limit - (standard + 1))
+    {
+      if (_pitch != 0)
+      {
+        if (index < limit - standard + standard / 2)
+          boundary_info.add_side(elem, 1, 3);
+        else
+          boundary_info.add_side(elem, 1, 4);
+      }
+      else
+      {
+        boundary_info.add_side(elem, 1, 3);
+      }
+    }
+    index++;
+    if (index == (num_nodes_boundary + counter * (standard + 1)) - 1)
+    {
+      index++;
+      counter++;
+    }
+  }
+
+  // This is to set boundary names.
+  boundary_info.sideset_name(1) = "left";
+  boundary_info.sideset_name(2) = "bottom";
+
+  if (_pitch == 0.0)
+    boundary_info.sideset_name(3) = "outer";
+  else
+  {
+    boundary_info.sideset_name(3) = "right";
+    boundary_info.sideset_name(4) = "top";
+  }
+
+  if (_portion == "top_left")
+  {
+    MeshTools::Modification::rotate(mesh, 90, 0, 0);
+    boundary_info.sideset_name(1) = "bottom";
+    boundary_info.sideset_name(2) = "right";
+
+    if (_pitch == 0.0)
+      boundary_info.sideset_name(3) = "outer";
+    else
+    {
+      boundary_info.sideset_name(3) = "top";
+      boundary_info.sideset_name(4) = "left";
+    }
+  }
+  else if (_portion == "bottom_left")
+  {
+    MeshTools::Modification::rotate(mesh, 180, 0, 0);
+    boundary_info.sideset_name(1) = "right";
+    boundary_info.sideset_name(2) = "top";
+
+    if (_pitch == 0.0)
+      boundary_info.sideset_name(3) = "outer";
+    else
+    {
+      boundary_info.sideset_name(3) = "left";
+      boundary_info.sideset_name(4) = "bottom";
+    }
+  }
+  else if (_portion == "bottom_right")
+  {
+    MeshTools::Modification::rotate(mesh, 270, 0, 0);
+    boundary_info.sideset_name(1) = "top";
+    boundary_info.sideset_name(2) = "left";
+
+    if (_pitch == 0.0)
+      boundary_info.sideset_name(3) = "outer";
+    else
+    {
+      boundary_info.sideset_name(3) = "bottom";
+      boundary_info.sideset_name(4) = "right";
+    }
+  }
+  else if (_portion == "top_half")
+  {
+    ReplicatedMesh & original_mesh = cast_ref<ReplicatedMesh &>(getMesh());
```

Comment:
you already have a reference to the `MeshBase` object called `mesh`. Couldn't you simply have `mesh` be of type `ReplicatedMesh` and cast it first thing in the `build` function. That makes this cast here redundant.